raiden/network/proxies/token.py

```diff
@@ -99,49 +111,23 @@ def approve(self, allowed_address: Address, allowance: TokenAmount) -> None:
                         block_identifier=block,
                     )
 
-                    msg = self._check_why_approved_failed(allowance, block)
-                    raise RaidenUnrecoverableError(f"{error_prefix}. {msg}")
-
-    def _check_why_approved_failed(
-        self, allowance: TokenAmount, block_identifier: BlockSpecification
-    ) -> str:
-        user_balance = self.balance_of(
-            address=self.client.address, block_identifier=block_identifier
-        )
-
-        # If the balance is zero, either the smart contract doesnt have a
-        # balanceOf function or the actual balance is zero
-        if user_balance == 0:
-            msg = (
-                "Approve failed. \n"
-                "Your account balance is 0 (zero), either the smart "
-                "contract is not a valid ERC20 token or you don't have funds "
-                "to use for openning a channel. "
-            )
-
-        # The approve call failed, check the user has enough balance
-        # (assuming the token smart contract may check for the maximum
-        # allowance, which is not necessarily the case)
-        elif user_balance < allowance:
-            msg = (
-                f"Approve failed. \n"
-                f"Your account balance is {user_balance}. "
-                f"The requested allowance is {allowance}. "
-                f"The smart contract may be rejecting your request due to the "
-                f"lack of balance."
-            )
-
-        # If the user has enough balance, warn the user the smart contract
-        # may not have the approve function.
-        else:
-            msg = (
-                f"Approve failed. \n"
-                f"Your account balance is {user_balance}. Nevertheless the call to"
-                f"approve failed. Please make sure the corresponding smart "
-                f"contract is a valid ERC20 token."
-            ).format(user_balance)
-
-        return msg
+                    balance = self.balance_of(self.client.address, checking_block)
```

Comment:
This should not use `checking_block`.

If the transaction failed when executing, we would check the receipt's block_hash (similar to [here](https://github.com/raiden-network/raiden/blob/7bc9b7c4e1b31b26e17ed225c3045d2b1fbf489d/raiden/network/proxies/token_network.py#L869)).
If the gas estimation failed, we use the latest block's blockhash (similar to [here](https://github.com/raiden-network/raiden/blob/7bc9b7c4e1b31b26e17ed225c3045d2b1fbf489d/raiden/network/proxies/token_network.py#L989)).